analytics/services.py

```diff
@@ -0,0 +1,16 @@
+from multiset.db_utils import execute_query
+from purchases.models import Purchase
+from purchases.services import get_purchases_by_group_id
+from collections import defaultdict
+
+
+def get_purchase_category_count(group_id: int):
+    rows = get_purchases_by_group_id(group_id)
```

Comment:
Should this logic be contained in the query instead? Ordinarily I'd agree w putting it here but db course yk

Reply:
Yes good suggestion! I just updated this